File: java-demo/src/main/java/zlc/season/rxdownload/java_demo/CustomSqliteActor.java
package zlc.season.rxdownload.java_demo;


import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;

import org.jetbrains.annotations.NotNull;

import zlc.season.rxdownload3.core.Mission;
import zlc.season.rxdownload3.core.RealMission;
import zlc.season.rxdownload3.database.SQLiteActor;

public class CustomSqliteActor extends SQLiteActor {
    private String IMG = "img";

    public CustomSqliteActor(@NotNull Context context) {
        super(context);
    }

    @NotNull
    @Override
    public String provideCreateSql() {
        String formatStr =
                "CREATE TABLE %s (\n" +
                        "%s TEXT PRIMARY KEY NOT NULL,\n" +
                        "%s TEXT NOT NULL,\n" +
                        "%s TEXT,\n" +
                        "%s TEXT,\n" +
                        "%s INTEGER,\n" +
                        "%s TEXT,\n" +
                        "%s TEXT,\n" +
                        "%s INTEGER,\n" +
                        "%s TEXT)";

        return String.format(formatStr, getTABLE_NAME(),
                getTAG(),
                getURL(),
                getSAVE_NAME(),
                getSAVE_PATH(),
                getRANGE_FLAG(),
                getCURRENT_SIZE(),
                getTOTAL_SIZE(),
                getSTATUS_FLAG(),
                IMG);
    }

    @NotNull
    @Override
    public ContentValues onCreate(@NotNull RealMission mission) {
        ContentValues cv = super.onCreate(mission);
        if (mission.getActual() instanceof CustomMission) {
            CustomMission customMission = (CustomMission) mission.getActual();
            cv.put(IMG, customMission.getImg());
        }
        return cv;
    }

    @NotNull
    @Override
    public Mission onGetAllMission(@NotNull Cursor cursor) {
        Mission mission = super.onGetAllMission(cursor);
        String img = cursor.getString(cursor.getColumnIndexOrThrow(IMG));
        if (img == null || img.isEmpty()) {
            img = "no img";
        }
        return new CustomMission(mission, img);
    }
}

File: kotlin-demo/src/main/java/zlc/season/rxdownload/kotlin_demo/CustomSqliteActor.kt
package zlc.season.rxdownload.kotlin_demo

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import zlc.season.rxdownload3.core.Mission
import zlc.season.rxdownload3.core.RealMission
import zlc.season.rxdownload3.core.Status
import zlc.season.rxdownload3.database.SQLiteActor


class CustomSqliteActor(context: Context) : SQLiteActor(context) {
    private val IMG = "img"

    override fun provideCreateSql(): String {
        return """
            CREATE TABLE $TABLE_NAME (
                $TAG TEXT PRIMARY KEY NOT NULL,
                $URL TEXT NOT NULL,
                $SAVE_NAME TEXT,
                $SAVE_PATH TEXT,
                $RANGE_FLAG INTEGER,
                $CURRENT_SIZE TEXT,
                $TOTAL_SIZE TEXT,
                $STATUS_FLAG INTEGER,
                $IMG TEXT)
            """
    }

    override fun onCreate(mission: RealMission): ContentValues {
        val cv = super.onCreate(mission)
        if (mission.actual is CustomMission) {
            val actualMission = mission.actual as CustomMission
            cv.put(IMG, actualMission.img)
        }

        return cv
    }

    override fun onGetAllMission(cursor: Cursor): Mission {
        val mission = super.onGetAllMission(cursor)
        var img = cursor.getString(cursor.getColumnIndexOrThrow(IMG))
        if (img.isNullOrEmpty()) {
            img = "no img"
        }
        return CustomMission(mission, img)
    }
}